client/my-sites/plans-features-main/index.jsx

```diff
@@ -36,8 +37,27 @@ import { plansLink } from 'lib/plans';
 import SegmentedControl from 'components/segmented-control';
 import SegmentedControlItem from 'components/segmented-control/item';
 import PaymentMethods from 'blocks/payment-methods';
+import HappychatButton from 'components/happychat/button';
+import HappychatConnection from 'components/happychat/connection-connected';
+import isHappychatAvailable from 'state/happychat/selectors/is-happychat-available';
+import { selectSiteId } from 'state/help/actions';
 
 class PlansFeaturesMain extends Component {
+	componentWillUpdate( nextProps ) {
+		/**
+		 * Happychat does not update with the selected site right now :(
+		 * This ensures that Happychat groups are correct in case we switch sites while on the plans
+		 * page, for example between a Jetpack and Simple site.
+		 *
+		 * @TODO: When happychat correctly handles site switching, remove selectSiteId action.
+		 */
+		const siteId = get( this.props, [ 'site', 'ID' ] );
+		const nextSiteId = get( nextProps, [ 'site', 'ID' ] );
+		if ( siteId !== nextSiteId && nextSiteId ) {
+			this.props.selectSiteId( nextSiteId );
```

Comment:
I'm not a big fan of the name `selectSiteId`. From a quick glance, it looks like it's an action for changing the selected site ID globally. But looking deeper, it's just for changing the selected site for Happychat. Could we maybe import it as something more descriptive, like `selectSiteIdForHappychat`?

Reply:
I agree, I'll swap for a clearer alternative.